packages/renderer/src/lib/ui/TerminalWindows.spec.ts

```diff
@@ -0,0 +1,162 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render } from '@testing-library/svelte';
+import { FitAddon } from '@xterm/addon-fit';
+import { Terminal } from '@xterm/xterm';
+import { writable } from 'svelte/store';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import TerminalWindow from '/@/lib/ui/TerminalWindow.svelte';
+
+import { TerminalSettings } from '../../../../main/src/plugin/terminal-settings';
+
+vi.mock('@xterm/xterm', () => ({
+  Terminal: vi.fn(),
+}));
+
+vi.mock('@xterm/addon-fit', () => ({
+  FitAddon: vi.fn(),
+}));
+
+const TerminalMock: Terminal = {
+  onWriteParsed: vi.fn(),
+  onResize: vi.fn(),
+  dispose: vi.fn(),
+  loadAddon: vi.fn(),
+  open: vi.fn(),
+  write: vi.fn(),
+} as unknown as Terminal;
+
+const FitAddonMock: FitAddon = {
+  fit: vi.fn(),
+} as unknown as FitAddon;
+
+beforeEach(() => {
+  vi.resetAllMocks();
+
+  vi.mocked(Terminal).mockReturnValue(TerminalMock);
+  vi.mocked(FitAddon).mockReturnValue(FitAddonMock);
+
+  Object.defineProperty(window, 'addEventListener', {
```

Comment:
I thought this was not required anymore

Reply:
AddEventListener is something in the default windows api, not something we mock through the IPC (my understanding is that we only mock the IPC we defined through electron)

Reply:
We seems to have a lot of tests which uses the logic of the event system, so we cannot mock it globally :/

Reply:
in that case, as the addEventListener already exists, we should "spy" and not replace the definition of the method by a mock

Reply:
I removed the define property and replaced it with a vi#spy